src/effects/VST3/VST3Effect.cpp

```diff
@@ -481,45 +473,23 @@ RegistryPaths VST3Effect::GetFactoryPresets() const
    return mFactoryPresets;
 }
 
-bool VST3Effect::LoadFactoryPreset(int id, EffectSettings &) const
+bool VST3Effect::LoadFactoryPreset(int id, EffectSettings& settings) const
 {
    if(id >= 0 && id < mFactoryPresets.size())
    {
       auto filename = wxFileName(GetFactoryPresetsPath(mEffectClassInfo), mFactoryPresets[id] + ".vstpreset");
       // To do: externalize state so const_cast isn't needed
-      return const_cast<VST3Effect*>(this)->LoadPreset(filename.GetFullPath());
+      const bool res = const_cast<VST3Effect*>(this)->LoadPreset(filename.GetFullPath());
+      FetchSettings(GetSettings(settings));
```

Comment:
I think two new calls to `FetchSettings` can simplify to one, if you lower it into `LoadPreset`, which will cover the import path too.